### PR TITLE
Mention VSCode and pylance

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

**Comment by srittau** [comment]
It's possible that we don't need to mention this, as it basically uses pyright under the hood.

**Review by rchen152 — COMMENTED** [review]
I think it's fine to mention these for completeness. I wonder if we should list pylance as "Pylance (language server extension)" just so readers know what it is - it's a bit of an odd duck that doesn't fit into any of the existing header categories.

File: pep-9999.rst [hunk 1]
```diff
@@ -993,14 +995,16 @@ PEPs
...
+Type Checkers and IDEs
+----------------------
 
 .. [#mypy] mypy -- Optional Static Typing for Python (http://www.mypy-lang.org/)
 .. [#pycharm] PyCharm -- The Python IDE for Professional Developers (https://www.jetbrains.com/pycharm/)
+.. [#pylance] Pylance -- Fast, feature-rich language support for Python (https://github.com/microsoft/pylance-release)
 .. [#pyre] Pyre -- A performant type-checker for Python 3 (https://pyre-check.org/)
 .. [#pyright] Pyright -- Static type checker for Python (https://github.com/microsoft/pyright)
 .. [#pytype] pytype -- A static analyzer for Python code (https://github.com/google/pytype)
+.. [#vscode] Visual Studio Code -- Code Editing. Redefined (https://code.visualstudio.com/)
```
Comment (Solumin) [inline pep-9999.rst:1011]: Is there a particular extension that's used with VS Code?

Comment (srittau) [reply pep-9999.rst:1011]: There is an official Python extension by Microsoft, but installing it is automatically suggested (and possible with one click) when opening a Python project. Do you think we should mention it separately?

Comment (srittau) [reply pep-9999.rst:1011]: And pylance is a separate extension that extends the extension. (I think, I haven't tried it.) Maybe that is something @jakebailey can clear up?

File: pep-9999.rst [hunk 2]
```diff
@@ -967,6 +967,8 @@ Development Environments
 
 * PyCharm [#pycharm]_, an IDE that supports type stubs both for type
   checking and code completion.
+* Visual Studio Code [#vscode]_, a code editor that supports type
+  checking using mypy, Pyright, or the Pylance [#pylance]_ extension.
```
Comment (jakebailey) [inline pep-9999.rst]: Nit: pyright is typically stylized lowercase like mypy.

File: pep-9999.rst (hunk 1 above)
Comment (jakebailey) [reply pep-9999.rst:1011]: Yes, Pylance is an extension for VS Code, so I'd probably mention them together if anything. Pylance hard depends on the Python extension, so you'd get both by installing Pylance. I wouldn't mention the Python extension alone, as it doesn't do type checking except via Pylance (or via linting with mypy or other tools).

**Pull request merged**